core/src/main/java/hudson/model/Job.java

```diff
@@ -1210,7 +1210,8 @@ public synchronized void doConfigSubmit(StaplerRequest req,
 
             String newName = req.getParameter("name");
             final ProjectNamingStrategy namingStrategy = Jenkins.getInstance().getProjectNamingStrategy();
-            if (newName != null && !newName.equals(name)) {
+            if (newName != null && !newName.trim().equals(name.trim())) {
```

Comment:
@KostyaSha perhaps you have a suggestion to make it 'prettier'? Rather than simply noting the fact you don't like something, providing a way to improve it would more easily result in it getting 'fixed'. For example, do you know of a utility method in Jenkins that would make this easier to understand?